worker/introspection/worker.go

```diff
@@ -302,3 +330,92 @@ func (a ValueSort) Less(i, j int) bool {
 	}
 	return a[i].ConnectionID < a[j].ConnectionID
 }
+
+type unitsHandler struct {
+	clock Clock
+	hub   Hub
+	done  <-chan struct{}
+}
+
+// ServeHTTP is part of the http.Handler interface.
+func (h unitsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if err := r.ParseForm(); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "%s\n", err)
+		return
+	}
+
+	switch action := r.Form.Get("action"); action {
+	case "":
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintln(w, "missing action")
+	case "start":
+		h.publishUnitsAction(w, r, "start", agent.StartUnitTopic)
+	case "stop":
+		h.publishUnitsAction(w, r, "stop", agent.StopUnitTopic)
+	case "status":
+		h.status(w, r)
+	default:
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "unknown action: %q\n", action)
+	}
+}
+
+func (h unitsHandler) publishUnitsAction(w http.ResponseWriter, r *http.Request, action string, topic string) {
+	if r.Method != http.MethodPost {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		fmt.Fprintf(w, "%s requires a POST request\n", action)
+		return
+	}
+
+	units := r.Form["unit"]
+	if len(units) == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintln(w, "missing unit")
+		return
+	}
+
+	h.hub.Publish(topic, agent.Units{Names: units})
+	if len(units) == 1 {
+		fmt.Fprintf(w, "requested unit %s to %s\n", units[0], action)
+	} else {
+		fmt.Fprintf(w, "requested units %s to %s\n", strings.Join(units, ", "), action)
```

Comment:
Should we wait for the response that they have been started.
That way any error will be printed to the user immediately, otherwise they have no idea if the request errorred and would need to look at logs

Reply:
Sure.